### PR TITLE
feat(server-hibernate): add flyway-postgresql dependency

### PR DESCRIPTION
**Pull request by eikenilsknopp**

### CODE REVIEW

File: sda-commons-server-hibernate/build.gradle [hunk 1]
```diff
@@ -16,6 +16,7 @@ dependencies {
   api 'org.flywaydb:flyway-core'
   api 'jakarta.annotation:jakarta.annotation-api'
   api 'org.flywaydb:flyway-core'
+  api 'org.flywaydb:flyway-database-postgresql'
```
Comment (JoergSiebahn) [inline sda-commons-server-hibernate/build.gradle:19]: Would it be enough to just add it to the dependency management and check if it works without version in an example module?

File: sda-commons-dependencies/build.gradle [hunk 2]
```diff
@@ -119,7 +120,8 @@ dependencies {
 
     // sda-commons-server-hibernate
     api "org.postgresql:postgresql:42.7.1"
-    api "org.flywaydb:flyway-core:10.7.2"
+    api "org.flywaydb:flyway-core:${flywayVersion}"
+    api "org.flywaydb:flyway-database-postgresql:${flywayVersion}"
```
Comment (JoergSiebahn) [inline sda-commons-dependencies/build.gradle:124]: Afaik SI is also using DB2 in base services. Should we manage that as well?

File: sda-commons-server-hibernate/build.gradle (hunk 1 above)
Comment (eikenilsknopp) [reply sda-commons-server-hibernate/build.gradle:19]: As Steve pointed out, the module is documented to [work specifically with PostgreSQL](https://github.com/SDA-SE/sda-dropwizard-commons/blob/ba635ebac3dd4d5885234a533d5c3ae07ff70095/docs/server-hibernate.md?plain=1#L6), so in my opinion it should provide everything needed to work out of the box.

File: sda-commons-dependencies/build.gradle (hunk 2 above)
Comment (eikenilsknopp) [reply sda-commons-dependencies/build.gradle:124]: We can, then I would add that to the documentation as well. At the moment it's just documented to work with postgre.

File: sda-commons-server-hibernate/build.gradle (hunk 1 above)
Comment (JoergSiebahn) [reply sda-commons-server-hibernate/build.gradle:19]: Hmmm, ok. Then let's keep it as it is here.

File: sda-commons-dependencies/build.gradle (hunk 2 above)
Comment (JoergSiebahn) [reply sda-commons-dependencies/build.gradle:124]: Let's just wait until we get a specific request for DB2. I hope, that you can pull in the Flyway DB2 (or other) module in addition to the Postgres module.

But regarding the docs: Maybe [this](https://github.com/SDA-SE/sda-dropwizard-commons/blob/ba635ebac3dd4d5885234a533d5c3ae07ff70095/docs/server-hibernate.md?plain=1#L13) is too specific. Could you add the flyway postgres module and remove the versions?

**Review by JoergSiebahn — APPROVED** [review]

**Pull request merged**